libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -90,358 +78,527 @@ string ASTJsonConverter::sourceLocationToString(SourceLocation const& _location)
+		{{"literals", literals}}
+	);
+	return false;
 }
 
 bool ASTJsonConverter::visit(ImportDirective const& _node)
 {
-	addJsonNode(_node, "ImportDirective", { make_pair("file", _node.path())});
-	return true;
+	std::vector<pair<string const, Json::Value>> attributes = {
+		make_pair("file", _node.path()),
+		make_pair("absolutePath", _node.annotation().absolutePath),
+		make_pair("SourceUnit", _node.annotation().sourceUnit->id()),
+		make_pair("scope", idOrNull(_node.scope()))
+	};
+	attributes.push_back(make_pair("unitAlias", _node.name()));
+	Json::Value symbolAliases(Json::arrayValue);
+	for (auto const& symbolAlias: _node.symbolAliases())
+	{
+		Json::Value tuple(Json::objectValue);
+		solAssert(symbolAlias.first, "");
+		tuple["foreign"] = symbolAlias.first->id();
+		tuple["local"] =  symbolAlias.second ? Json::Value(*symbolAlias.second) : Json::nullValue;
+		symbolAliases.append(tuple);
+	}
+	attributes.push_back( make_pair("symbolAliases", symbolAliases));
+	setJsonNode(_node, "ImportDirective", std::move(attributes));
+	return false;
 }
 
-bool ASTJsonConverter::visit(ContractDefinition const& _node)
+bool ASTJsonConverter::visit(ContractDefinition const& _node) //todo add inheritancespecifier
 {
 	Json::Value linearizedBaseContracts(Json::arrayValue);
 	for (auto const& baseContract: _node.annotation().linearizedBaseContracts)
 		linearizedBaseContracts.append(Json::UInt64(baseContract->id()));
-	addJsonNode(_node, "ContractDefinition", {
+	Json::Value contractDependencies(Json::arrayValue);
+	for (auto const& dependentContract: _node.annotation().contractDependencies)
+		contractDependencies.append(Json::UInt64(dependentContract->id()));
+	setJsonNode(_node, "ContractDefinition", {
 		make_pair("name", _node.name()),
 		make_pair("isLibrary", _node.isLibrary()),
 		make_pair("fullyImplemented", _node.annotation().isFullyImplemented),
 		make_pair("linearizedBaseContracts", linearizedBaseContracts),
-	}, true);
-	return true;
+		make_pair("contractDependencies", contractDependencies),
+		make_pair("nodes", toJson(_node.subNodes())),
+		make_pair("scope", idOrNull(_node.scope()))
+	});
+	return false;
 }
 
 bool ASTJsonConverter::visit(InheritanceSpecifier const& _node)
 {
-	addJsonNode(_node, "InheritanceSpecifier", {}, true);
-	return true;
+
+	//??this node never shows up!
+	cout << "imin" << endl; //apparently this function is not even entered....
+	// assumed usage:
+	// import contract.sol <- with bar-contract
+	// contract foo is bar {...
+	setJsonNode(_node, "InheritanceSpecifier", {
+		make_pair("baseName", toJson(_node.name())),
+		make_pair("arguments", toJson(_node.arguments()))
+	});
+	return false;
 }
 
 bool ASTJsonConverter::visit(UsingForDirective const& _node)
 {
-	addJsonNode(_node, "UsingForDirective", {}, true);
-	return true;
+	setJsonNode(_node, "UsingForDirective", {
+		make_pair("libraryNames", toJson(_node.libraryName())),
+		make_pair("typeName", _node.typeName() ? toJson(*_node.typeName()) : Json::Value("*"))
```

Comment:
I know I probably said something else before, but I guess it is better to use `null` instead of `"*"` because of "type-safety": As it is now, you either have an object or a specific string, but I think it would be more natural to either have an object or `null`.